### PR TITLE
Add testclient unit tests

### PR DESCRIPTION
**Pull request by xfix**
Not meant to be merged, essentially I want to have this pull request so that I could run Travis tests (feature involves Travis), the feature is not finished. Comments welcome however.

### CODE REVIEW

**Comment by xfix** [comment]
Hm, but this will be too limited for say battle engine tests where you may want to challenge other player and so on...

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,21 @@
 sudo: false
 language: node_js
+dist: trusty
 node_js:
-  - "4"
+  - "7.7"
```
Comment (Zarel) [inline .travis.yml]: Can you use Node 8 here?

**Comment by Zarel** [comment]
Why not? Tell me when you want this merged.

**Comment by xfix** [comment]
Why is Travis still not checking this, it was a hour or so. Weird.

**Comment by Zarel** [comment]
Maybe you're requiring an Ubuntu version that's hard to provision or something?

**Comment by xfix** [comment]
Anyway, this should be fine to merge now.

File: package.json [hunk 2]
```diff
@@ -10,11 +10,13 @@
   },
   "scripts": {
     "test": "eslint --config=.eslintrc.js --cache --cache-file=eslint-cache/base js/*.js data/graphics.js && eslint --config=githooks/.eslintrc.js --cache --cache-file=eslint-cache/build githooks/update githooks/build-indexes",
-    "fix": "eslint --config=.eslintrc.js --fix js/*.js data/graphics.js && eslint --config=githooks/.eslintrc.js --fix githooks/update githooks/build-indexes"
+    "fix": "eslint --config=.eslintrc.js --fix js/*.js data/graphics.js && eslint --config=githooks/.eslintrc.js --fix githooks/update githooks/build-indexes",
+    "testcafe": "githooks/build-indexes && testcafe chrome test/"
```
Comment (Zarel) [inline package.json]: Why is this not a part of `test`?

Comment (xfix) [inline package.json]: Because it clones Pokemon Showdown repository and opens a web browser without closing it.

**Pull request merged**